### PR TITLE
Bugfix: Return correct container Ids for multiple container stats

### PR DESCRIPTION
**Pull request by wlan0**
@cjellick @vincent99

### CODE REVIEW

File: stats/aggregated_stats.go [hunk 1]
```diff
@@ -23,12 +23,9 @@ func convertToAggregatedStats(id string, containerIds map[string]string, resourc
 		aggregatedStats := []AggregatedStat{}
 		for _, stat := range stats {
 			if len(stat.Stats) > i {
-				if resourceType == "container" && id == "" {
-					id = stat.Name
-				}
```
Comment (wlan0) [inline stats/aggregated_stats.go]: setting the id here resulted in all container stats getting the same containerId, as I was not resetting the id to "" after using the id.

File: stats/aggregated_stats.go [hunk 2]
```diff
@@ -23,12 +23,9 @@ func convertToAggregatedStats(id string, containerIds map[string]string, resourc
...
-				statPoint := convertCadvisorStatToAggregatedStat(id, containerIds, resourceType, memLimit, stat.Stats[i])
+				statPoint := convertCadvisorStatToAggregatedStat(id, stat.Name, containerIds, resourceType, memLimit, stat.Stats[i])
 				if statPoint.Id == "" {
-					return totalAggregatedStats
+					continue
```
Comment (wlan0) [inline stats/aggregated_stats.go]: returning the complete stats when one of the Ids is not found would result in getting no stats at all, as there will be an empty id on every loop from the rancher-agent container.

**Comment by wlan0** [comment]
Fixes https://github.com/rancher/rancher/issues/1938

**Comment by cjellick** [comment]
Per my comment in slack. Let's get #45 merged and then have you write tests for this before we merge it.

File: stats/integration_test.go [hunk 3]
```diff
@@ -20,9 +23,132 @@ import (
...
+	}
+	defer ws.Close()
+
+	for count := 0; count < 4; count++ {
+		_, msg, err := ws.ReadMessage()
+		if err != nil {
+			t.Fatal(err)
+		}
+		stats := string(msg)
+		if !strings.Contains(stats, "1i1") || !strings.Contains(stats, "1i2") {
```
Comment (cjellick) [inline stats/integration_test.go]: The way this is written, it doesn't actually test the bug we're fixing, right? To fully assert the fix, we'd have to ensure that we see both 1i1 and 1i2, not either.

File: scripts/test [hunk 4]
```diff
@@ -23,6 +23,9 @@ fi
 
 set_project_vars
 
+#a dummy cadvisor for testing multiple container stats
+docker run --detach=true google/cadvisor:latest
```
Comment (cjellick) [inline scripts/test]: This won't scale well as we add more tests to host-api (something I really want to do now that we have a start).
I think the right thing to do is to start and destroy the test containers in the acttual test. The paradigm established in python-agent is a good one to follow: each test is responsible for creating the container(s) it needs to test the feature.

Comment (wlan0) [inline scripts/test]: sgtm.

File: stats/integration_test.go (hunk 3 above)
Comment (wlan0) [inline stats/integration_test.go]: This tests that both 1i1 and 1i2 exists. Not either.

Comment (wlan0) [inline stats/integration_test.go]: This tests that both 1i1 and 1i2 exists, not either.

Comment (cjellick) [inline stats/integration_test.go]: Ah, I was thinking about it somewhat backwards. nevermind

**Comment by cjellick** [comment]
LGTM

**Pull request merged**